tests/ga/test_update.py

```diff
@@ -2110,6 +2110,60 @@ class ReportStatusTestCase(AgentTestCase):
     """
     Tests for UpdateHandler._report_status()
     """
+
+    def setUp(self):
+        self.patches = [
+            patch("time.sleep", side_effect=lambda _: mock_sleep(0.001)),
+            patch("sys.exit")
+        ]
+
+        for p in self.patches:
+            p.start()
+
+        return AgentTestCase.setUp(self)
+
+    def tearDown(self):
+        
+        for p in self.patches:
+            p.stop()
+
+        return AgentTestCase.tearDown(self)
+
+    def test_upload_status_blob_even_on_failed_goal_state(self):
+
+        @contextlib.contextmanager
+        def mock_update_handler():
+            def goal_state_handler(url, **kwargs): # pylint: disable=unused-argument
```

Comment:
NIT: remove pylint supression

Reply:
I believe this function's signature has to coincide with `http_request` which takes in info not needed by this function. I'll add a comment mentioning why this supression is necessary.